pyroomacoustics/libroom_src/tests/test_ray_tracing.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+import pyroomacoustics as pra
+from pyroomacoustics.utilities import fractional_delay
+import matplotlib.pyplot as plt
+import time
+import scipy
+from scipy.io import wavfile
+from scipy import signal
+
+wall_corners_strange = [
+    np.array([  # front
+        [0, 3, 3, 0],
+        [0, 0, 0, 0],
+        [0, 0, 2, 2],
+    ]),
+    np.array([  # back
+        [0, 0, 6, 6],
+        [8, 8, 8, 8],
+        [0, 4, 4, 0],
+    ]),
+    np.array([  # floor
+        [0, 0, 6, 3, ],
+        [0, 8, 8, 0, ],
+        [0, 0, 0, 0, ],
+    ]),
+    np.array([  # ceiling
+        [0, 3, 6, 0, ],
+        [0, 0, 8, 8, ],
+        [2, 2, 4, 4, ],
+    ]),
+    np.array([  # left
+        [0, 0, 0, 0, ],
+        [0, 0, 8, 8, ],
+        [0, 2, 4, 0, ],
+    ]),
+    np.array([  # right
+        [3, 6, 6, 3, ],
+        [0, 8, 8, 0, ],
+        [0, 0, 4, 2, ],
+    ]),
+]
+
+wall_corners_cube = [
+    np.array([  # front
+        [0, 10, 10, 0],
+        [0, 0, 0, 0],
+        [0, 0, 10, 10],
+    ]),
+    np.array([  # back
+        [0, 10, 10, 0],
+        [10, 10, 10, 10],
+        [0, 0, 10, 10],
+    ]),
+    np.array([  # floor
+        [0, 0, 10, 10, ],
+        [0, 10, 10, 0, ],
+        [0, 0, 0, 0, ],
+    ]),
+    np.array([  # ceiling
+        [0, 0, 10, 10, ],
+        [0, 10, 10, 0, ],
+        [10, 10, 10, 10, ],
+    ]),
+    np.array([  # left
+        [0, 0, 0, 0, ],
+        [0, 0, 10, 10, ],
+        [0, 10, 10, 0, ],
+    ]),
+    np.array([  # right
+        [10, 10, 10, 10, ],
+        [0, 10, 10, 0, ],
+        [0, 0, 10, 10, ],
+    ]),
+]
+
+absorptions = [0.1]*len(wall_corners_strange)
+
+
+def test_room_construct(cube):
+
+    # Choose which room to use
+    if cube :
+        walls = [pra.libroom_new.Wall(c, a) for c, a in zip(wall_corners_cube, absorptions)]
+        obstructing_walls = []
+        microphones = np.array([
+            [6., ],
+            [8., ],
+            [5., ],
+        ])
+
+        room = pra.libroom_new.Room(walls, obstructing_walls, microphones)
+
+    else :
+        walls = [pra.libroom_new.Wall(c, a) for c, a in zip(wall_corners_strange, absorptions)]
+        obstructing_walls = []
+        microphones = np.array([
+            [2.5, ],
+            [7, ],
+            [1., ],
+        ])
+
+        room = pra.libroom_new.Room(walls, obstructing_walls, microphones)
+
+    return room
+
+def compute_rir(log, time_thres, fs, plot=True):
```

Comment:
You can use the fast RIR builder here:
```
def compute_rir(log, time_thres, fs, plot=True):


    TIME = 0
    ENERGY = 1

    time = np.array([e[TIME] for e in log], dtype=np.float)
    alpha = np.array([e[ENERGY] for e in log], dtype=np.float)
    visibility = np.ones(time.shape[0], dtype=np.int32)

    # ======= WITH FRACTIONAL PART =======

    # the python utilities to compute the rir
    fdl = pra.constants.get('frac_delay_length')
    fdl2 = (fdl - 1) // 2  # Integer division

    ir = np.zeros(int(time_thres * fs) + fdl)

    from pyroomacoustics import build_rir
    build_rir.fast_rir_builder(ir, time, alpha, visibility, fs, fdl)

    if plot :
        x = np.arange(len(ir)) / fs
        plt.figure()
        plt.plot(x, ir)
        plt.title("RIR")
        plt.show()

    return ir
```

Reply:
I have a problem with this function. 
The time parameter is defined as follows : 

`time: ndarray (double)
        The array of delays for the image sources`

And then there is an assertion to check that `floor(fs * np.min(time)) - fdl2 >= 0`

But for raytracing, min(time) is 0, so this statement is always wrong :/

Reply:
Yes, that's a problem: the Ray tracer should not give minimum time of zero. This would only happen if the source were exactly at the same location as the mic. This problem comes from the output vector having empty entries when you create it. I have tried this part and I could solve it by setting the size of output vector to zero at the beginning the algorithm.